core/dbt/config/runtime.py

```diff
@@ -243,7 +261,7 @@ def validate(self):
     def collect_parts(cls: Type["RuntimeConfig"], args: Any) -> Tuple[Project, Profile]:
         # profile_name from the project
         project_root = args.project_dir if args.project_dir else os.getcwd()
-        cli_vars: Dict[str, Any] = getattr(args, "vars", {})
+        cli_vars: Dict[str, Any] = parse_cli_vars(getattr(args, "vars", "{}"))
```

Comment:
[parse_cli_vars done during parsing in click](https://github.com/dbt-labs/dbt-core/compare/feature/click-cli#diff-abc64babd717570b5e76349ece5b307aa645315415db68fef2c14d782e1e058cR17-R18) instead of as-necessary on main.  Restoring `parse_cli_vars` usage here and elsewhere on this branch to get backwards compatibility with the old framework for the purposes of testing.